# What: Fix compiler error of address XXX will never be NULL.
#       This patch is based on https://github.com/oneapi-src/oneTBB/pull/1181
# Error msg:
#           .../doctest.h:1560:42: error: the compiler can assume that the address of doctest::detail::Expression_lhs<const char (&)[58]>::lhs will never be NULL [-Werror=address]
#           1560 |             bool res = static_cast<bool>(lhs);
# Author: maxim-masterov

diff -Nru oneTBB-2021.10.0.orig/test/common/doctest.h oneTBB-2021.10.0/test/common/doctest.h
--- oneTBB-2021.10.0.orig/test/common/doctest.h	2023-09-08 13:28:11.213989000 +0200
+++ oneTBB-2021.10.0/test/common/doctest.h	2023-09-08 13:29:01.734231041 +0200
@@ -1555,10 +1555,13 @@
                 , m_at(at) {}
 
         DOCTEST_NOINLINE operator Result() {
+// TODO: upstream the change to doctest : Work-around for the warning: 'address will never be NULL'
+DOCTEST_GCC_SUPPRESS_WARNING_WITH_PUSH("-Waddress")
 // this is needed only for MSVC 2015
 DOCTEST_MSVC_SUPPRESS_WARNING_WITH_PUSH(4800) // 'int': forcing value to bool
             bool res = static_cast<bool>(lhs);
 DOCTEST_MSVC_SUPPRESS_WARNING_POP
+DOCTEST_GCC_SUPPRESS_WARNING_POP
             if(m_at & assertType::is_false) { //!OCLINT bitwise operator in conditional
                 res = !res;
             }
